src/pages/workspaces/workspace/analysis/file-utils.ts

```diff
@@ -52,11 +59,15 @@ export interface AnalysisFile {
   metadata?: AnalysisFileMetadata
 }
 
+export type CreateAnalysisFn = (fullAnalysisName: string, toolLabel: ToolLabel, contents: any) => Promise<void>
+
 export interface AnalysisFileStore {
-  refresh: () => Promise<void>
+  refreshFileStore: () => Promise<void>
   loadedState: LoadedState<AnalysisFile[], unknown>
-  create: (fullAnalysisName: string, toolLabel: ToolLabel, contents: string) => Promise<void>
+  create: CreateAnalysisFn
+  deleteFile: (path: AbsolutePath) => Promise<void>
   pendingCreate: LoadedState<true, unknown>
+  pendingDelete: LoadedState<true, unknown>
 }
 
 export const useAnalysisFiles = (): AnalysisFileStore => {
```

Comment:
this hook (and the related bits) should be split out to it's own useAnalysisFiles.ts file.  This is a great candidate to chip away at the dumping-grounds-utils files problem we have.